### PR TITLE
Solution setup

### PR DESCRIPTION
**Pull request by scottluskcis**
addressing #1 in part for setup of DI 

- setup .NET Core DI and got working with injecting a DbContext stub and the IUserManager and IDbRepository implementations 
- removed Autofac, switched to .NET Core built in DI 
- added stub ApplicationDbContext
- added an AboutController and set it as default launchUrl, api/values did not exist 
- can run /users route now and see that IUserManager is injected appropriately, tracing through the stack you'll see the in memory application db context as well 
- more work to add but this is a start for working with Dependency Injection

### CODE REVIEW

**Comment by tylerbhughes** [comment]
This may be because I'm on a Mac currently I'm having issues when I run `dotnet restore`.

```
  Restore completed in 74.61 ms for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Core/UrbanEngine.Core.csproj.
  Restore completed in 79.23 ms for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Infrastructure/UrbanEngine.Infrastructure.csproj.
  Restore completed in 3.38 ms for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Tests/UrbanEngine.Tests.csproj.
  Restore completed in 53.83 ms for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Tests/UrbanEngine.Tests.csproj.
  Restoring packages for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj...
/Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj : warning NU1608: Detected package version outside of dependen
cy constraint: Microsoft.AspNetCore.App 2.1.0 requires Microsoft.EntityFrameworkCore (= 2.1.0) but version Microsoft.EntityFrameworkCore 2.1.2 was res
olved. [/Users/tyler/Documents/projects/urban-engine-api/urban-engine-api.sln]
/Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj : error NU1107: Version conflict detected for Microsoft.Entity
FrameworkCore.Abstractions. Reference the package directly from the project to resolve this issue.  [/Users/tyler/Documents/projects/urban-engine-api/
urban-engine-api.sln]
/Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj : error NU1107:  UrbanEngine.Web -> Microsoft.EntityFrameworkC
ore 2.1.2 -> Microsoft.EntityFrameworkCore.Abstractions (>= 2.1.2)  [/Users/tyler/Documents/projects/urban-engine-api/urban-engine-api.sln]
/Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj : error NU1107:  UrbanEngine.Web -> Microsoft.AspNetCore.App 2.1.0 -> Microsoft.EntityFrameworkCore.Abstractions (= 2.1.0). [/Users/tyler/Documents/projects/urban-engine-api/urban-engine-api.sln]
  Restore failed in 1.74 sec for /Users/tyler/Documents/projects/urban-engine-api/UrbanEngine.Web/UrbanEngine.Web.csproj.
```

Have you run into this?

**Comment by scottluskcis** [comment]
I did not run into that. I took the nuget package that was there already and just added it to the Web project without any issues. I'll try to do a restore to see if I encounter same problem

File: UrbanEngine.Web/Controllers/AboutController.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
+﻿using System.Reflection; 
+using Microsoft.AspNetCore.Mvc;
+
+namespace UrbanEngine.Web.Controllers {
+    [Route( "api/[controller]" )]
```
Comment (tylerbhughes) [inline UrbanEngine.Web/Controllers/AboutController.cs]: Per our conversation, the API will be hosted at the subdomain api.urbanengine.org so I don't think the `api/` is needed here.

File: UrbanEngine.Web/Properties/launchSettings.json [hunk 2]
```diff
@@ -12,15 +12,15 @@
     "IIS Express": {
       "commandName": "IISExpress",
       "launchBrowser": true,
-      "launchUrl": "api/values",
+      "launchUrl": "api/about",
```
Comment (tylerbhughes) [inline UrbanEngine.Web/Properties/launchSettings.json]: `api/` should be removed here as well

File: UrbanEngine.Web/Properties/launchSettings.json [hunk 3]
```diff
@@ -12,15 +12,15 @@
...
       "environmentVariables": {
         "ASPNETCORE_ENVIRONMENT": "Development"
       }
     },
     "urban_engine_api": {
       "commandName": "Project",
       "launchBrowser": true,
-      "launchUrl": "api/values",
+      "launchUrl": "api/about",
```
Comment (tylerbhughes) [inline UrbanEngine.Web/Properties/launchSettings.json]: `api/` should be removed here as well

Comment (scottluskcis) [reply UrbanEngine.Web/Properties/launchSettings.json]: updated

File: UrbanEngine.Web/Controllers/AboutController.cs (hunk 1 above)
Comment (scottluskcis) [reply UrbanEngine.Web/Controllers/AboutController.cs]: updated

File: UrbanEngine.Web/Properties/launchSettings.json (hunk 2 above)
Comment (scottluskcis) [reply UrbanEngine.Web/Properties/launchSettings.json]: updated

**Comment by scottluskcis** [comment]
sorry @tylerbhughes I did a push upstream and it merged this automatically, I'll avoid that in the future

**Pull request merged**